### DISCUSSION TITLE
User-scripts versus addons/extensions

### DISCUSSION THREAD

**Discussion by J-HS-L** [insin/control-panel-for-twitter#137] [Q&A] [answered]
Are there any considerable differences, between the two? 
E.g. perhaps the update cadence is different...

Currently, I'm using the userscript...
But wondering if best to use the Edge (Chromium-based) add-on/extension.

**Comment by J-HS-L** [comment]
@insin, @brechtcs, @THEDARKK, @rogama25, or anyone?

Thanks again!

**Comment by insin** [comment, marked as answer]
User scripts always get updates first, as there's no review/publishing delay - I can just put the latest release straight onto Greasy Fork.

Other than that, it's mostly negatives for user scripts: you have to edit your config preferences by hand, which then get blown away if you update using your user script manager, whereas extensions give you a nice real-time options popup and your preferences aren't affected by updates.

Personally, I use the Firefox extension, but if Twitter makes a breaking change I disable the extension and re-enable the user script version while I'm waiting for the fixed Firefox version to be published (which has been very slow of late).

**Reply by J-HS-L** [reply]
@insin 

Thanks, mate!

> ...which then get blown away if you update using your user script manager

Yeah, but some of them now have a built-in BU feature, don't they?! Then again;
I don't think any of them are automated, & who remembers to do that manually. 

:D
